src/Hosting/Abstractions/src/IHostApplicationLifetime.cs

```diff
@@ -16,14 +16,14 @@ public interface IHostApplicationLifetime
         CancellationToken ApplicationStarted { get; }
 
         /// <summary>
-        /// Triggered when the application host is performing a graceful shutdown.
-        /// Shutdown will block until this event completes.
+        /// Triggered when the application host is beginning a graceful shutdown.
+        /// Shutdown will block until any callbacks registered on this token have returned.
         /// </summary>
         CancellationToken ApplicationStopping { get; }
 
         /// <summary>
-        /// Triggered when the application host is performing a graceful shutdown.
-        /// Shutdown will block until this event completes.
+        /// Triggered when the application host has completed a graceful shutdown.
+        /// The application will not exit until any callbacks registered on this token have returned.
```

Comment:
```suggestion
        /// The application will not exit until all callbacks registered on this token have completed.
```

nit: any -> all, returned -> completed.